packages/cms-cli/lib/commonOpts.js

```diff
@@ -43,9 +51,16 @@ const setLogLevel = (options = {}) => {
   }
 };
 
+/**
+ * Obtains portalId using supplied --portal flag or from environment variables
+ */
 const getPortalId = (options = {}) => {
-  const { portal } = options;
-  return getPortalIdFromConfig(portal);
+  const envConfig = loadConfigFromEnvironment();
+  const { portal: portalName } = options;
+  return (
+    (envConfig && envConfig.portals[0].portalId) ||
+    getPortalIdFromConfig(portalName)
+  );
```

Comment:
Instead of a complex one liner here, let's have an early exit. I think it also is reasonable for the environment variable to always take precedence.
```suggestion
  if (process.env.HUBSPOT_PORTAL_ID) {
    return process.env.HUBSPOT_PORTAL_ID;
  }
  return getPortalIdFromConfig(portalName);
```

Reply:
This is the code that I previously had, but I changed it because if the `HUBSPOT_PORTAL_ID` is set, but not any of the other necessary environment variables needed to successfully load a config, we run into an error here because we are using the `HUBSPOT_PORTAL_ID` for `portalId` and the rest of the portal config is loaded from file.

Reply:
Under what circumstances does the error occur?

Reply:
[Loading an environment variable config](https://github.com/HubSpot/hubspot-cms-tools/blob/886b154c23187c3c856f9741db163724a4f35739/packages/cms-lib/lib/config.js#L542) requires the `HUBSPOT_PORTAL_ID` plus at least one other environment variable. If only `HUBSPOT_PORTAL_ID` is present and I use the code above, `getPortalIdFromConfig` will use the portalId from the environment variable even though the config being used is being loaded from the `hubspot.config.yml` file, which can cause issues depending on where `getPortalId` is being called from.

By doing the check using `loadConfigFromEnvironment` we are verifying that the required environment variables to load a config are present, so we are in fact loading the config from the environment variables as opposed to the config file.

Reply:
I still would like to make this more straightforward. The `allowEnvironmentVariableConfig` is awkward to me because it feels like we should be able to handle the two scenarios in a more direct than indirect fashion. It isn't clear why something that is loading the config should need to be configured such that under some circumstances it should check environment variables and another it shouldn't.

From a user, it also isn't clear why I couldn't do `HUBSPOT_PORTAL_ID=123 hs fetch foo bar` if the portal has been configured. It is clear why `HUBSPOT_PORTAL_ID=123 hs fetch --portal 456 foo bar` should throw and error. And, it also is clear why certain sets of env variables shouldn't be allowed.

Reply:
> can cause issues depending on where getPortalId is being called from.

What kinds of issues? I ask because the function is part of CLI and used to process the command options, so I am not sure what scenarios might be a concern.

Reply:
> I still would like to make this more straightforward. The `allowEnvironmentVariableConfig` is awkward to me because it feels like we should be able to handle the two scenarios in a more direct than indirect fashion. It isn't clear why something that is loading the config should need to be configured such that under some circumstances it should check environment variables and another it shouldn't.

I replaced `allowEnvironmentVariableConfig` with `ignoreEnvironmentVariableConfig` which only needs to be set by `hs auth` in order to allow writes to file when there are environment variables present when running the command.

> From a user, it also isn't clear why I couldn't do `HUBSPOT_PORTAL_ID=123 hs fetch foo bar` if the portal has been configured. It is clear why `HUBSPOT_PORTAL_ID=123 hs fetch --portal 456 foo bar` should throw and error. And, it also is clear why certain sets of env variables shouldn't be allowed.

I updated the way this works so that setting `HUBSPOT_PORTAL_ID` works similarly to `defaultPortal`.